File: be/internal/model/audit.go
package model

import "time"

// AuditEntry represents a row in the audit_log table.
type AuditEntry struct {
	ID           string    `json:"id"`
	UserID       string    `json:"user_id,omitempty"` // empty if user deleted
	Action       string    `json:"action"`
	ResourceType string    `json:"resource_type"`
	ResourceID   string    `json:"resource_id"`
	IP           string    `json:"ip"`
	UserAgent    string    `json:"user_agent"`
	Metadata     string    `json:"metadata"` // JSON object
	CreatedAt    time.Time `json:"created_at"`
}

// AuditFilter restricts audit_log queries.
type AuditFilter struct {
	UserID string
	Action string
	Since  *time.Time
	Until  *time.Time
}

File: ui/src/types/audit.ts
export interface AuditEntry {
  id: string
  user_id?: string
  action: string
  resource_type: string
  resource_id: string
  ip: string
  user_agent: string
  metadata: string
  created_at: string
}

export interface AuditListResponse {
  items: AuditEntry[]
  total: number
  page: number
  per_page: number
}
